### PR TITLE
Waterfall Graph

### PR DESCRIPTION
**Pull request by pldespaigne**

### CODE REVIEW

File: libs/ui/src/lib/graph/graph.component.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import { BehaviorSubject } from 'rxjs';
+
+
+@Component({
+  selector: 'bf-graph-container',
+  templateUrl: './graph.component.html',
+  styleUrls: ['./graph.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export default class GraphContainerComponent implements AfterViewInit {
```
Comment (phpgeekfr) [inline libs/ui/src/lib/graph/graph.component.ts:12]: to stick with what we are used to :

```suggestion
export  class GraphContainerComponent implements AfterViewInit {
```
in module
```
import { GraphContainerComponent }  from './graph.component';
```

File: libs/ui/src/lib/graph/graph.component.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  dragging = false;
+
+  public offset = new BehaviorSubject({ x: 0, y: 0, z: 1 });
+
+  @ViewChild('dragContainer') container: ElementRef<HTMLDivElement>;
+
+  ngAfterViewInit(
+  ) {
+    this.offset.subscribe(offset => {
```
Comment (phpgeekfr) [inline libs/ui/src/lib/graph/graph.component.ts:22]: you should unsubscribe on ngDestroy

File: libs/waterfall/src/lib/components/graph/graph.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  select(id: string) {
+    this.selected$.next(id);
+  }
+
+  async layout() {
+    const { nodes, arrows } = await toGraph(this.rights, this.sources, this.state);
+    this.nodes$.next(nodes);
+    this.arrows$.next(arrows);
+  }
```
Comment (phpgeekfr) [inline libs/waterfall/src/lib/components/graph/graph.component.ts:43]: you can put this in ngOnInit directly

File: libs/waterfall/src/lib/components/graph/horizontal/horizontal.module.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+@NgModule({
+  declarations: [ WaterfallGraphHorizontalComponent ],
+  imports: [
+    CommonModule,
+
+    WaterfallGraphRightModule,
+    WaterfallGraphVerticalModule,
+
+    MatTooltipModule,
```
Comment (phpgeekfr) [inline libs/waterfall/src/lib/components/graph/horizontal/horizontal.module.ts:21]: looks like you don't use it

**Review by phpgeekfr — APPROVED** [review]

File: libs/waterfall/src/lib/components/graph/graph.component.ts (hunk 3 above)
Comment (pldespaigne) [reply libs/waterfall/src/lib/components/graph/graph.component.ts:43]: Might need to recompute the layout during editing once the form is implemented

**Pull request merged**